src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,307 @@
+     * An array of unacceptable parent of ast of type {@link TokenTypes#IDENT}.
+     */
+    private static final int[] UNACCEPTABLE_PARENT_OF_IDENT = {
+        TokenTypes.DOT,
+        TokenTypes.LITERAL_NEW,
+        TokenTypes.METHOD_CALL,
+        TokenTypes.TYPE,
+    };
+
+    /**
+     * Keeps tracks of the current catch parameter.
+     */
+    private CatchParameter catchParameter;
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_CATCH,
+            TokenTypes.IDENT,
+        };
+    }
+
+    @Override
+    public void beginTree(DetailAST rootAST) {
+        catchParameter = null;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (ast.getType() == TokenTypes.LITERAL_CATCH) {
+            visitCatchParameter(ast);
+        }
+        else {
+            visitIdent(ast);
+        }
+    }
+
+    @Override
+    public void leaveToken(DetailAST ast) {
+        logUnusedCatchParameter(ast);
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#LITERAL_CATCH}
+     * and set the catch parameter.
+     *
+     * @param catchAst token representing {@link TokenTypes#LITERAL_CATCH}
+     */
+    private void visitCatchParameter(DetailAST catchAst) {
+        final DetailAST catchParameterAst = catchAst.findFirstToken(TokenTypes.PARAMETER_DEF);
+        final String catchParameterName =
+                catchParameterAst.findFirstToken(TokenTypes.IDENT).getText();
+
+        if (!"_".equals(catchParameterName)) {
+            catchParameter = new CatchParameter(catchParameterName,
+                                    catchAst, catchParameterAst);
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#IDENT}
+     * and check if it is a candidate for a catch parameter usage.
+     *
+     * @param identAST token representing {@link TokenTypes#IDENT}
+     */
+    private void visitIdent(DetailAST identAST) {
+        final boolean isCatchParameter = identAST.getParent().getParent()
+                                                .getType() == TokenTypes.LITERAL_CATCH;
+
+        if (!isCatchParameter
+                && (!TokenUtil.isOfType(identAST.getParent(), UNACCEPTABLE_PARENT_OF_IDENT)
+                        || shouldCheckDotChild(identAST))) {
+            checkIdentifier(identAST);
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     * and should be checked for usage.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     *     and should be checked for usage.
+     */
+    private static boolean shouldCheckDotChild(DetailAST identAst) {
```

Comment:
What is a dotChild? Like, is this a qualified name?

Reply:
something like. if the ident is the first child of the dot(left of it), then it is a candidate for catch parameter, if it is not (right of dot) then it can't be catch parameter ident, and it is anything else like `this.e` or `super.e`

```
`--DOT -> . [7:35]
     |--LITERAL_THIS -> this [7:31]
     `--IDENT -> e [7:36]        <-- not first child, can't be a candidate for the catch parameter
```

```
 |--DOT -> . [7:13]
     |--IDENT -> e [7:12]  first child, a candidate for the catch parameter
     `--IDENT -> printStackTrace [7:14]
```

Reply:
We need to name this something meaningful, like `isMethodInvocation``?